File: Python3/3280_convert_date_to_binary.py
# https://leetcode.cn/problems/convert-date-to-binary/

class Solution:
    def convertDateToBinary(self, date: str) -> str:
        def to_binary(s):
            res = ""
            num = int(s)
            while num:
                res += str(num & 1)
                num >>= 1
            res = res[::-1]
            return res

        res = []
        ymd = date.split("-")
        for s in ymd:
            res.append(to_binary(s))
        return "-".join(res)

File: Cpp/3280_convert_date_to_binary.cpp
// https://leetcode.cn/problems/convert-date-to-binary/

class Solution {
public:
    string convertDateToBinary(string date) {
        string res = "";
        res += to_binary(date.substr(0, 4));
        res += '-';
        res += to_binary(date.substr(5, 2));
        res += '-';
        res += to_binary(date.substr(8, 2));
        return res;
    }

    string to_binary(string s) {
        string res = "";
        int num = stoi(s);
        while (num) {
            res += '0' + (num & 1);
            num >>= 1;
        }
        reverse(res.begin(), res.end());
        return res;
    }
};
